src/server/util/meta_tree.cc

```diff
@@ -405,46 +406,36 @@ static void generate_put_ops(const ptree& meta, const ptree& diff,
   }
 }
 
-static void generate_persist_ops(const ptree& diff, const std::string& name,
+/// Store metas to etcd.
+/// We create an entry for each subtree recursively, and reset processed
+/// subtree to its link to avoid duplicate information.
+static void generate_persist_ops(ptree& diff, const std::string& name,
                                  std::vector<IMetaService::op_t>& ops,
                                  std::set<std::string>& dedup) {
   std::string key_prefix = "data." + name + ".";
-  for (ptree::const_iterator it = diff.begin(); it != diff.end(); ++it) {
+  for (ptree::iterator it = diff.begin(); it != diff.end(); ++it) {
```

Comment:
Why it cannot be `const iterator`?

Reply:
It modified the tree inplace (delete processed subtree to avoid duplication)